### DISCUSSION TITLE
Controller is missing `defaultValue` prop when using `useFieldArray`

### DISCUSSION THREAD

**Discussion by oltodo** [react-hook-form/react-hook-form#4193] [General] [answered]
Hi, I'm getting warnings when using `Controller` component through a `useFieldArray` hook.

Consider the basic example below:

https://codesandbox.io/s/thirsty-bardeen-jn6iy?fontsize=14&hidenavigation=1&theme=dark

```jsx
import { TextField } from "@material-ui/core";
import { Controller, useFieldArray, useForm } from "react-hook-form";
import "./styles.css";

export default function App() {
  const { control } = useForm({
    defaultValues: {
      labels: [{ text: "Lorem" }, { text: "Ipsum" }, { text: "Dolor" }]
    }
  });
  const { fields } = useFieldArray({
    name: "labels",
    control
  });

  return (
    <div className="App">
      <h1>
        Warnings with <code>useFieldArray</code>
      </h1>

      {fields.map((item, index) => (
        <Controller
          key={item.id}
          name={`labels[${index}].text`}
          control={control}
          render={({ value, ref, onChange, onBlur }) => (
            <TextField
              label="Title"
              variant="filled"
              margin="dense"
              value={value}
              inputRef={ref}
              onChange={onChange}
              onBlur={onBlur}
              fullWidth
            />
          )}
        />
      ))}
    </div>
  );
}
```

When the form is opening, I can see 3 text fields containing 3 default values (respectively `Lorem`, `Ipsum` and `Dolor`). But I get the following warning:

> Controller is missing `defaultValue` prop when using `useFieldArray`. https://react-hook-form.com/api#Controller

So when I add a `defaultValue=""` prop to `<Controller>` the warnings disappear but the text fields stay empty.

Do you have an idea about what am I supposed to do to get my text fields set up with their default values without getting these warnings?

**Comment by bluebill1049** [comment, marked as answer]
`defaultValue={field.text}` will set the correct default value for the input.

**Reply by oltodo** [reply]
So `Controller` can't only be based on `defaultValues` passed to `useForm` in the same way as when there is no `defaultValue` set?

**Reply by bluebill1049** [reply]
in FieldArray you will always need to supply a defaultValue, because the fields will get shuffled.
